Assets/Scripts/Utils/GenericTwoBodyIK.cs

```diff
@@ -0,0 +1,53 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class GenericTwoBodyIK : MonoBehaviour
+{
+    /// <summary>
+    /// Manually sets the rotation of all 3 given transforms that will be animated by IK.
+    /// Transfoms should have a capsule collider to define length, else localScale magnitude will be used (which is less accurate)
+    /// 
+    /// Note: Due to scaling of world distances compared to localscale, 
+    /// one might have to tweak the localScaleOffset to get proper bending, usually in the range of [0.5] or smaller
+    /// </summary>
+    /// <param name="root">First bone</param>
+    /// <param name="joint">Second bone, will extend towards pole when squished</param>
+    /// <param name="end">Third and final bone, will always point to target</param>
+    /// <param name="pole">The half-sphere direction which the second bone points from</param>
+    /// <param name="target">The goal point of the end bone</param>
+    /// <param name="rootOffset">Optional extra rotation of root around its IK forward axis</param>
+    /// <param name="jointOffset">optinal extra rotation of joint around its IK forward axis</param>
+    /// <param name="localScaleOffset">Optional relative scaling for achieving distances comparable </param>
+    public static void AnimateTransforms(Transform root, Transform joint, Transform end, Transform pole, Transform target, float rootOffset = 0, float jointOffset = 0, float localScaleOffset = 1)
```

Comment:
So you ended up writing your own IK code, then? Unity didn't have anything that fit?

Reply:
Nope, they have a 2bodyIK component, but that only works with humanoid rigs (Our player uses a generic rig), and for some reason Unity has no "generic" alternative 🙃.